### ISSUE TITLE
Bug: Grouped Curtains Returning No Status, cannot close

### ISSUE THREAD

**Issue by jeremybell** [OpenWonderLabs/homebridge-switchbot#650]
### Describe The Bug

After grouping my curtains in the Switchbot app, they will always been seen as closed in Homebridge.

### To Reproduce

Group 2 curtain bots in the Switchbot app (making them appear as 1). Add the device in Homebridge, and attempt to open the curtains. Nothing will happen. However, you can close them.

### Expected behavior

They should open/close when commanded, like a good robot.

### Relevant log output

```shell
[1/13/2023, 11:20:56 AM] [SwitchBot] Total SwitchBot Devices Found: 3
[1/13/2023, 11:20:56 AM] [SwitchBot] Adding new accessory: Living Room Curtains Curtain DeviceID: FD90B0540EA4
[1/13/2023, 11:20:56 AM] [SwitchBot] Curtain: Living Room Curtains Config: {"set_minStep":25,"setOpenMode":"0","setCloseMode":"0","hide_lightsensor":true,"connectionType":"OpenAPI"}
[1/13/2023, 11:22:20 AM] [SwitchBot] Curtain: Living Room Curtains Checking Status ...
```

### Config for homebridge-switchbot

```shell
{
            "name": "SwitchBot",
            "credentials": {
                "token": "XXXXXXXXXXXXXXXXXXXXXXXX",
                "secret": "XXXXXXXXXXXXXXXXXXXXXXXX",
                "notice": "Keep your Token & Secret a secret!"
            },
            "options": {
                "devices": [
                    {
                        "deviceId": "FD90B0540EA4",
                        "configDeviceType": "Curtain",
                        "connectionType": "OpenAPI",
                        "curtain": {
                            "set_minStep": 25,
                            "setOpenMode": "0",
                            "setCloseMode": "0",
                            "hide_lightsensor": true
                        },
                        "disableCaching": true
                    }
                ]
            },
            "_bridge": {
                "username": "0E:D1:7D:62:3F:71",
                "port": 33837
            },
            "platform": "SwitchBot"
        }
```

### Screenshots

![IMG_0718](https://user-images.githubusercontent.com/541144/212371602-90a40dd4-54ea-48bf-97c0-ad9e1a0f7983.PNG)

### Device & Model

Hub Mini, Curtain Bot

### Node.js Version

v18.13.0

### NPM Version

v9.3.0

### Homebridge Version

v1.6.0

### Homebridge Switchbot Plugin Version

v2.4.1-beta.7

### Homebridge Config UI X Plugin Version

v4.50.2

### Operating System

Docker

**Comment by jeremybell** [comment]
I have installed the latest beta, cleared the homebridge cache, reset the tokens, re-added all of the devices, and changed various config settings... nothing seems to work.

**Comment by jeremybell** [comment]
I've tried upgrouping and resetting everything, and still no luck. I've cleared the caches and removed the devices from HomeBridge. I've created sub-bridges and that didn't do anything either. 

I can see the devices, but I cannot close them. They always appear as closed in HomeKit. They WILL close when commanded, but they won't open (and there is never a updated status in the logs)

**Comment by donavanbecker** [comment]
Where these curtains in Homebridge before you grouped them?

**Comment by jeremybell** [comment]
I don't recall. They were definitely in HK beforehand, but they had stopped working. I only discovered SwitchBot supported grouping while I was troubleshooting why they had stopped working.

I've since removed and re-added them in HK, plus I've entirely removed and re-added the HomeBridge plugin... nothing seems to help. 

I'm on the new HomeKit architecture. Unsure this is this part of the mystery. But the blinds had originally stopped working well before I upgraded.

**Comment by donavanbecker** [comment]
Can you try a previous version of the plugin?

**Comment by jeremybell** [comment]
I went back to v2.3.1 and have the same results. In the logs, it only says "Curtains checking status..." but it never receives a response. 

In the "accessories" section in Homebridge, when I tap the curtains accessory, it will open the curtains (and say it's opening)... but it will immediately toggle back to the closed state after a few seconds. 

I also just tried rebooting the hub itself (as I realized I hadn't earlier). Didn't help. Same result.

**Comment by donavanbecker** [comment]
There is a disable cache config. Can you try that?

**Comment by dnicolson** [comment]
I checked the "Disable Caching" checkbox under "SwitchBot Device Settings" but it had no effect. As mentioned in https://github.com/OpenWonderLabs/homebridge-switchbot/pull/658, Homebridge needs to be restarted for the state to update.

#652 and #653 appear to be duplicates of this.

**Comment by nyjklein** [comment]
I think my bug #666 is also the same as this. In my case the grouped curtains have been working essentially forever. This issue just started recently.

**Comment by jeremybell** [comment]
I've tried disabling cache, and nothing seems to help. However, I enable debug logging, and it looks to be a timeout issue?

```
[2/6/2023, 4:24:34 PM] [SwitchBot] [DEBUG] Curtain: Curtains openAPIpushChanges statusCode: 200
[2/6/2023, 4:24:34 PM] [SwitchBot] [DEBUG] Curtain: Curtains Unknown statusCode.
[2/6/2023, 4:24:34 PM] [SwitchBot] [DEBUG] Curtain: Curtains d: {"statusCode":100,"body":{"items":[{"deviceID":"FD90B0540EA4","code":100,"status":{"position":0},"message":"ok"}]},"message":"success"}
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains Refresh Status When Moving, PositionState: 1
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains openAPIRefreshStatus
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains sign: qiIFaCLGaVPMuiooftJgwlkR9eR6CM4FFSAJQomR6zc=
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains statusCode: 200
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains d: {"statusCode":100,"body":{"deviceId":"FD90B0540EA4","deviceType":"Curtain","hubDeviceId":"E7044AF56DE3","slidePosition":100},"message":"success"}
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains refreshStatus: {"statusCode":100,"body":{"deviceId":"FD90B0540EA4","deviceType":"Curtain","hubDeviceId":"E7044AF56DE3","slidePosition":100},"message":"success"}
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains openAPIparseStatus
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain Curtains CurrentPosition: 0
[2/6/2023, 4:24:35 PM] [SwitchBot] Curtain: Curtains Checking Status ...
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains Standby, CurrentPosition: 0
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains Stopped
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains CurrentPosition: 0, TargetPosition: 0, PositionState: 2,
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic CurrentPosition: 0
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic PositionState: 2
[2/6/2023, 4:24:35 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic TargetPosition: 0
[2/6/2023, 4:24:40 PM] [SwitchBot] [DEBUG] Curtain: Curtains setNewTarget true timeout
[2/6/2023, 4:24:48 PM] [SwitchBot] [DEBUG] Curtain: Curtains openAPIRefreshStatus
[2/6/2023, 4:24:48 PM] [SwitchBot] [DEBUG] Curtain: Curtains sign: dUlx08Qqi6Sa8km0O5MNo9qw7zOlcnmV/2XP7E6+e6Q=
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains statusCode: 200
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains d: {"statusCode":100,"body":{"deviceId":"FD90B0540EA4","deviceType":"Curtain","hubDeviceId":"E7044AF56DE3","slidePosition":100},"message":"success"}
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains refreshStatus: {"statusCode":100,"body":{"deviceId":"FD90B0540EA4","deviceType":"Curtain","hubDeviceId":"E7044AF56DE3","slidePosition":100},"message":"success"}
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains openAPIparseStatus
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain Curtains CurrentPosition: 0
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains Standby, CurrentPosition: 0
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains Stopped
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains CurrentPosition: 0, TargetPosition: 0, PositionState: 2,
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic CurrentPosition: 0
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic PositionState: 2
[2/6/2023, 4:24:49 PM] [SwitchBot] [DEBUG] Curtain: Curtains updateCharacteristic TargetPosition: 0
```

Note this line in particular: 

`[2/6/2023, 4:24:40 PM] [SwitchBot] [DEBUG] Curtain: Curtains setNewTarget true timeout`

**Comment by jeremybell** [comment]
I can manually set a specific % in Homekit (eg. move the slider to 2%), and it'll work just fine (that is, the curtains will close to 2%). However, in the Home app, it'll show them as "closed".

Tapping on the button in Home will then open the curtains, but the status will again revert back to "closed".

**Comment by jeremybell** [comment]
so it seems like perhaps something is timing out, which then reverts the state of the curtains as "closed".

**Comment by jeremybell** [comment]
> I think my bug #666 is also the same as this. In my case the grouped curtains have been working essentially forever. This issue just started recently.

For context, my grouped curtains were also working flawlessly previously as well (that is, "grouped" in Homekit, and not grouped via SwitchBot). I had originally thought this was an issue with how SwitchBot was handling grouping in their app (and therefore what is exposed to Homebridge), but I don't think it's related anymore. 

I simply cannot get my curtains to reliably work at all now, grouped or not. They will open, but the status will always revert back to "closed"

**Comment by nyjklein** [comment]
Out of curiosity, what version of tvOS/iOS are you running on your Homekit hub?

**Comment by jeremybell** [comment]
All available hubs (Apple TV or HomePod) are running 16.3. That said, these issues started before I upgraded the hubs. In fact, I upgraded to 16.3 in hopes that it would resolve them!

**Comment by donavanbecker** [comment]
duplicate to #666
